### ISSUE TITLE
Lack of root causes confusing error

### ISSUE THREAD

**Issue by jakubgs** [openzfs/zfs#12915]
### System information
Type | Version/Name
 --- | ---
Distribution Name	| NixOS
Distribution Version	| 21.11
Kernel Version	| `5.15.10`
Architecture	| `aarch64`
OpenZFS Version	| `2.1.2-1`

### Describe the problem you're observing
I was trying to create a mirror pool the other day on two drives in my new [ICYBOX IB-RD2253-U31](https://icybox.de/en/product.php?id=304) USB 3.1 enclosure connected to my [NanoPC-T4](https://www.friendlyarm.com/index.php?route=product/product&product_id=225), but when I did I got this weird error:
```
 > zpool create -f USB-HDD mirror sda sdb
the kernel failed to rescan the partition table: 13
cannot label 'sda': try using parted(8) and then provide a specific slice: -1
```
After 20 minutes of pulling my hair out I realized my stupid mistake and added `sudo`, which allowed it to create the pool without issues.

If I try to create a pool without `sudo` on a loopback device created from a file using `dd` and `losetup` the error makes sense though:
```
 > zpool create test /dev/loop0
cannot create 'test': permission denied
```
Not sure why it's different for the USB enclosure, but the error confused me quite a bit.

**Comment by Dragon-debian** [comment]
I dont think that a big problem; I mean its obvious that you have to run `zpool` with root privileges to create a pool. 
almost every time I want to do something related to partitions and disks, first I run `partprobe` to inform the kernel of latest partition table changes.

**Comment by GregorKopka** [comment]
> I dont think that a big problem; I mean its obvious that you have to run `zpool` with root privileges to create a pool. almost every time I want to do something related to partitions and disks, first I run `partprobe` to inform the kernel of latest partition table changes.

Would be nice to detect such missing privileges, to deliver an error message that stops people from chasing that kind of red herring.

**Comment by jakubgs** [comment]
Yep, a permission issue should cause a `permission denied` error to appear, not something about rescans, labels, or slices.

Not a single word in that error message indicates in any way an issue with permissions.

**Comment by jakubgs** [comment]
I don't think this is stale.

**Comment by jakubgs** [comment]
Nope, not stale.
